src/studio/src/designer/frontend/ux-editor/containers/EditContainer.tsx

```diff
@@ -225,14 +225,13 @@ export function EditContainer(props: IEditContainerProvidedProps) {
     setListItem(newListItem);
     setIsEditMode(false);
 
-    const { id: stateId, ...restState } = component;
-    const { id: idProps, ...restProps } = props.component;
-    if (JSON.stringify(restState) !== JSON.stringify(restProps)) {
+    if (JSON.stringify(component) !== JSON.stringify(props.component)) {
       handleSaveChange(component);
+      if (props.id !== component.id) {
+        dispatch(FormLayoutActions.updateFormComponentId({ newId: component.id, currentId: props.id }));
```

Comment:
I'm guessing this should be removed?

Reply:
I kept it, because we have a watcher for it for when we need to update conditional rendering rules with new id's.

Reply:
Ah, right! Good catch.